charon/pkgs/maven.py

```diff
@@ -497,6 +613,199 @@ def _scan_paths(files_root: str, ignore_patterns: List[str],
     return (top_level, valid_mvn_paths, valid_poms, valid_dirs)
 
 
+def _generate_rollback_archetype_catalog(
+    s3: S3Client, bucket: str,
+    root: str, prefix: str = None
+) -> int:
```

Comment:
I think this should return `-> Tuple[int, List[str]]` so we can have a list of files to act on (whether to delete or upload).